Objects/longobject.c

```diff
@@ -217,15 +217,30 @@ PyObject *
 _PyLong_Copy(PyLongObject *src)
 {
     assert(src != NULL);
+    int sign;
 
     if (_PyLong_IsCompact(src)) {
         stwodigits ival = medium_value(src);
         if (IS_SMALL_INT(ival)) {
             return get_small_int((sdigit)ival);
         }
+        sign = _PyLong_CompactSign(src);
+    }
+    else {
+        sign = _PyLong_NonCompactSign(src);
     }
+
     Py_ssize_t size = _PyLong_DigitCount(src);
-    return (PyObject *)_PyLong_FromDigits(_PyLong_IsNegative(src), size, src->long_value.ob_digit);
+    PyLongObject *result = long_alloc(size);
+
+    if (result == NULL) {
+        PyErr_NoMemory();
```

Comment:
This call is redundant, long_alloc() already calls PyErr_NoMemory(). Please remove it here but also in _PyLong_FromDigits().